### ISSUE TITLE
Redirect breaks routing with forward-slash `/`

### ISSUE THREAD

**Issue by peacechen** [dwyl/hapi-error#63]
I'm using the [redirect](https://github.com/dwyl/hapi-error#redirectredirecting-to-another-endpoint) feature which results in a 404 error.  The issue seems to be that the forward-slash `/` in the query parameter confuses Hapi's routing:
```
http://localhost:4000/login?redirect=/admin
```

There's a route for `/login` but the above doesn't match.  The slash causes it to look for another path `/login?redirect=/admin` despite it having a query parameter.

PR #55 could work around it by allowing me to URL-encode the redirect path.  That hasn't been merged though.

Edit:  more from the console error in case it helps.  Suspicious about the CORS error.
> OPTIONS http://localhost:4000/login?redirect=/admin 404 (Not Found)
> Access to fetch at 'http://localhost:4000/login?redirect=/admin' (redirected from 'http://localhost:4000/login') from origin 'http://localhost:3000' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.

**Comment by nelsonic** [comment]
@peacechen thank you for opening this issue to bring it to everyone's attention.
PR #55 is has not passed continuous integration tests and has merge conflicts.
If you have time to re-create the PR and ensure it passes, we would very gladly accept it.

**Comment by peacechen** [comment]
That PR has quite a few commits.  Would you mind granting me permission to squash them together and resolve conflicts?  Re-creating them all individually will take a long while.

I need to continue to debug the CORS error in my environment.  It may be a separate issue.

**Comment by DavidHavl** [comment]
The PR #55 has now been updated and conflicts resolved.

**Comment by nelsonic** [comment]
PR #55 merged. New version on NPM `2.3.0` contains the fix. :shipit:
